src/ArcCommander/APIs/GitAPI.fs

```diff
@@ -25,6 +17,40 @@ module GitAPI =
 
         Fake.IO.Path.getDirectory(gitDir)
 
+    /// Authenticates to a token service and stores the token using git credential manager.
+    let authenticate (arcConfiguration : ArcConfiguration) =
+
+        let log = Logging.createLogger "GitAuthenticateLog"
+
+        log.Info("Start Arc Authenticate")
+        
+        match Authentication.tryLogin log arcConfiguration with 
+        | Ok token -> 
+            log.Info($"INFO: Successfully retrieved access token from token service")
+
+            log.Trace($"TRACE: Try transfer git user metadata to global arcCommander config")
+            match IniData.tryGetGlobalConfigPath () with
+            | Some globalConfigPath ->
+                IniData.setValueInIniPath globalConfigPath "general.gitname"    (token.FirstName + " " + token.LastName)
+                IniData.setValueInIniPath globalConfigPath "general.gitemail"   token.Email
+                log.Trace($"TRACE: Successfully transferred git user metadata to global arcCommander config")
+            | None ->
+                log.Error($"ERROR: Could not transfer git user metadata to global arcCommander config")
+
+            if storeCredentialsToken log token then
+                log.Info($"Finished Authentication")
+
+            else
```

Comment:
What is the reason for using several log messages? For normal, we use one log message for one event:
```f#
log.Error("ERROR: Authentication worked, but credentials could not be stored successfully.\nCheck if git is installed and if a credential helper is setup:\nRun \"git config --global credential.helper cache\" to cache credentials in memory\nor Run \"git config --global credential.helper store\" to save credentials to disk\nFor more info go to: https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage")
```